asus/zephyrus/ga402x/shared.nix

```
@@ -5,7 +5,11 @@
 }:
 
 let
-  inherit (lib) mkDefault mkIf mkMerge version versionOlder;
+  inherit (lib) mkDefault mkEnableOption mkIf mkMerge version versionAtLeast versionOlder;
+
+  cfg = config.hardware.asus.zephyrus.ga402x;
+  defaultAutosuspendEnable = if (versionAtLeast version "6.9") then true else false;
```

you're testing the nixos version, not the kernel version
```suggestion
  defaultAutosuspendEnable = versionAtLeast config.boot.kernelPackages.kernel.version "6.9";
```